bpm/bonita-core/bonita-process-engine/src/main/java/org/bonitasoft/engine/api/impl/ServerAPIImpl.java

```diff
@@ -470,13 +465,20 @@ protected Object invokeAPI(final Object[] parametersValues, final Object apiImpl
         }
     }
 
+    private void warnIfDeprecated(Method method, String apiInterfaceName) {
+        if (technicalLogger != null && method.isAnnotationPresent(Deprecated.class)) {
+            technicalLogger.log(this.getClass(), TechnicalLogSeverity.WARNING, "The API method " + apiInterfaceName + "." + method.getName()
+                    + " is deprecated. It will be deleted in a future release. Please plan to update your code to use the replacement method instead. Check the Javadoc for more details.");
```

Comment:
Would be nice but we do not really know when deprecated method will be ffectivly removed.
Maybe having it at least in the javadoc the version in which it will be removed can be usefull when we know it?